### ISSUE TITLE
Plus (+) symbols are no longer being decoded into spaces when part of a URL capture

### ISSUE THREAD

**Issue by ghazel** [sinatra/sinatra#1004]
Regression of #463 caused by 4a02a757fde29ba21d241c838c8114f64a393fae

```ruby
2.1.5 :001 > require 'uri'
 => true 
2.1.5 :002 > URI::Parser.new.unescape("foo+bar")
 => "foo+bar" 
```

**Comment by Arcovion** [comment]
That commit message mentions:
> It also gets rid of the following warning when running in verbose mode:

> lib/sinatra/base.rb:972:in `block in process_route': warning: URI.unescape is obsolete

However `URI::unescape` is simply an alias for `URI::decode`.

It should use `CGI::unescape`, failing that `Rack::Utils::unescape` works too (which is just `URI::decode_www_form_component`).

**Comment by Arcovion** [comment]
Actually, this was the commit responsible: https://github.com/sinatra/sinatra/commit/b4d7d4bb1088f5a6caf1bfbbba80717db9232fe7
See the discussion here: https://github.com/sinatra/sinatra/issues/638

I'm having trouble understanding [this code](https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L880) included in the commit you referenced:
The URI [source code](https://github.com/ruby/ruby/blob/trunk/lib/uri/common.rb#L123-L126) shows that `URI::unescape` calls `DEFAULT_PARSER`, which is already `URI::Parser.new` as [shown here](https://github.com/ruby/ruby/blob/trunk/lib/uri/common.rb#L17-L21). So as far as I can tell all this does is avoid the deprecation warning...

Regardless of how the decoding should be handled, all uses of `URI` in Sinatra are for these deprecated methods. That should probably be addressed in future, along with the `URI_INSTANCE` bit.

**Comment by kytrinyx** [comment]
This should be fixed in master after merging #1018 today.
